homeassistant/components/synology_srm/device_tracker.py

```diff
@@ -75,9 +103,21 @@ def scan_devices(self):
 
     def get_extra_attributes(self, device) -> dict:
         """Get the extra attributes of a device."""
-        filter_result = [result for result in self.devices if result["mac"] == device]
-        if filter_result:
-            return filter_result[0]
+        device = next(
+            iter([result for result in self.devices if result["mac"] == device]), None
```

Comment:
```suggestion
            (result for result in self.devices if result["mac"] == device), None
```

Reply:
I am quite sure that you need iter on a list (I have always found this quite silly)?

Reply:
We're swapping the list comprehension for a generator expression. So no need to user `iter` then.